packages/x-charts/src/BarChart/useBarChartProps.ts

```diff
@@ -0,0 +1,162 @@
+import useId from '@mui/utils/useId';
+import type { BarChartProps } from './BarChart';
+import { DEFAULT_X_AXIS_KEY, DEFAULT_Y_AXIS_KEY } from '../constants';
+
+/**
+ * A helper function that extracts BarChartProps from the input props
+ * and returns an object with props for the children components of BarChart.
+ *
+ * @param props The input props for BarChart
+ * @returns An object with props for the children components of BarChart
+ */
+export const useBarChartProps = <T extends BarChartProps>(props: T) => {
```

Comment:
My idea was to allow properly typing something like `const {...rest} = useBarChartProps(proProps)` but might be kinda useless, since we could simply do `useBarChartProProps(props)` instead 😅